Bottom: 8cd3b8a3623cea41bb7c61500be99a544acca116
Top:    f894cfe10947313ebffb37f733b37d80a9f5ad72
Author: =?UTF-8?q?Emilio=20L=C3=B3pez?= <emilio@elopez.com.ar>
Date:   2014-03-24 21:28:34 -0300

From 330cb198f30065ac8852fe99159f1789f322ef36 Mon Sep 17 00:00:00 2001
Subject: [PATCH 04/19] Travis CI configuration


---

diff --git a/.travis.yml b/.travis.yml
new file mode 100644
index 0000000..e7c242c
--- /dev/null
+++ b/.travis.yml
@@ -0,0 +1,34 @@
+language: c
+
+compiler:
+  - gcc
+
+env:
+  - ARCH=arm LOADADDR=0x40008000 CROSS_COMPILE="arm-linux-gnueabihf-" BUILD_CONFIG=sunxi_defconfig TARGETS="uImage dtbs" J=1 C=2
+  - ARCH=arm LOADADDR=0x40008000 CROSS_COMPILE="arm-linux-gnueabihf-" BUILD_CONFIG=sunxi_full_defconfig TARGETS="uImage dtbs" J=1 C=2
+  - ARCH=arm LOADADDR=0x40008000 CROSS_COMPILE="arm-linux-gnueabihf-" BUILD_CONFIG=multi_v7_defconfig TARGETS="uImage dtbs modules" J=1 C=2
+  - BUILD_CONFIG=allyesconfig TARGETS="bzImage" J=8 C=0
+  - BUILD_CONFIG=allnoconfig TARGETS="bzImage" J=8 C=0
+  - BUILD_CONFIG=allmodconfig TARGETS="bzImage modules" J=8 C=0
+
+before_install:
+  - sudo sed -i 's/^.*rabbit.*$//' /etc/apt/sources.list
+  - sudo apt-get update -qq
+
+install:
+  - sudo apt-get install -qq bc sparse
+  - sh -c '[ -n "$ARCH" ] && sudo apt-get install -qq u-boot-tools gcc-arm-linux-gnueabihf || exit 0'
+
+before_script:
+  - ${CROSS_COMPILE}gcc -v
+  - make $BUILD_CONFIG
+
+script:
+  - make -j$J C=$C CONFIG_DEBUG_SECTION_MISMATCH=y $TARGETS
+
+notifications:
+  email:
+    recipients:
+      - travis@elopez.com.ar
+    on_success: change
+    on_failure: always
